src/plugins/intel_gpu/src/graph/graph_optimizer/mark_shape_of_subgraphs.cpp

```diff
@@ -78,6 +79,13 @@ bool mark_shape_of_subgraphs::can_mark_node(const program_node& node) {
             return false;
     }
 
+    // Exclude stride_slice primitive if it's input is big const ternsor, else CPU reference implementation
+    // will lead to huge performance drop.
+    if (node.is_type<strided_slice>() && node.get_dependency(0).is_constant() &&
+        node.get_dependency(0).get_output_layout().count() > 1024 * 1024) {
+        return false;
```

Comment:
I faced same issue for 7b model too, but the shape is 128*4K. 
Reducing the criteria to fit that shape could improve 7b model too. 
Initially using the cpu impl for shape of subgraph is assuming that the size is very small (i.e., like some sub portion of shapes). So I think we can reduce the criteria more, 128*1024 .. or even just 1024, so that we can improve 7b model too.